dotnet/src/Planners/Planners.OpenAI/Utils/EmbeddedResource.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.IO;
+using System.Reflection;
+using Microsoft.SemanticKernel.Diagnostics;
+
+#pragma warning disable IDE0130
+namespace Microsoft.SemanticKernel.Planners;
+#pragma warning restore IDE0130
+
+internal static class EmbeddedResource
+{
+    private static readonly string? s_namespace = typeof(EmbeddedResource).Namespace;
+
+    internal static string Read(string name)
+    {
+        var assembly = typeof(EmbeddedResource).GetTypeInfo().Assembly;
+        if (assembly == null) { throw new SKException($"[{s_namespace}] {name} assembly not found"); }
```

Comment:
Can you check is there a standard .Net exception we should use here

Reply:
`FileNotFoundException` seems like the closest for this and the comment below.

I copied the `EmbeddedResource` class from `Planners.Core` (and I see there's also one in `Experimental.Orchestration.Flow` so I will update the exceptions in those files as well.